x-pack/plugins/timelines/public/components/t_grid/toolbar/fields_browser/categories_selector.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import { omit } from 'lodash';
+import {
+  EuiFilterButton,
+  EuiFilterGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHighlight,
+  EuiPopover,
+  EuiSelectable,
+  FilterChecked,
+} from '@elastic/eui';
+import { BrowserFields } from '../../../../../common';
+import * as i18n from './translations';
+import { CountBadge, getFieldCount, CategoryName, CategorySelectableContainer } from './helpers';
+import { isEscape } from '../../../../../common/utils/accessibility';
+
+interface CategoriesSelectorProps {
+  /**
+   * A map of categoryId -> metadata about the fields in that category,
+   * filtered such that the name of every field in the category includes
+   * the filter input (as a substring).
+   */
+  filteredBrowserFields: BrowserFields;
+  /**
+   * Invoked when the user clicks on the name of a category in the left-hand
+   * side of the field browser
+   */
+  setSelectedCategoryIds: (categoryIds: string[]) => void;
+  /** The category selected on the left-hand side of the field browser */
+  selectedCategoryIds: string[];
+}
+
+interface CategoryOption {
+  label: string;
+  count: number;
+  checked?: FilterChecked;
+}
+
+const renderOption = (option: CategoryOption, searchValue: string) => {
+  const { label, count, checked } = option;
+  return (
+    <EuiFlexGroup
+      data-test-subj={`categories-selector-option-${label}`}
+      alignItems="center"
+      gutterSize="none"
+      justifyContent="spaceBetween"
+    >
+      <EuiFlexItem grow={false}>
+        <CategoryName
+          data-test-subj={`categories-selector-option-name-${label}`}
+          bold={checked === 'on'}
```

Comment:
`useMemo` for this please

Reply:
I can't use hooks inside this `renderOption` function, I would have to turn it into a component and pass a proxy function to the `renderOption` prop of `EuiSelectable`. I can do that but I'm not sure if it is worth it, considering it is just a boolean, it will only cause a re-render if the actual value changes. what do you think?

Reply:
yep just leave it then thanks